build-system/task-runner/install-amp-task-runner.js

```diff
@@ -34,7 +34,7 @@ const ampCliRunner = 'build-system/task-runner/amp-cli-runner.js';
  * @return {Promise<void>}
  */
 async function installAmpTaskRunner() {
-  const npmBinDir = getStdout('npm bin --global').trim();
```

Comment:
I did some more homework on this topic and remembered that we install the `amp` CLI runner to the global npm directory so that if a developer has multiple copies of the `amphtml` repo on their machine, the same global CLI runner copy (which is available on the system path once npm is installed) is used to invoke the repo-specific task.

If the `chown` workaround doesn't work elegantly, you could also try overriding the global npm prefix directory to a writable location on the docker container using `npm config set prefix path/to/writable/directory` before invoking `npm ci`.

Coupled with https://github.com/ampproject/amphtml/pull/35176#discussion_r668292374, I think there's good reason to reinstate `install_dependencies.sh` and distill an efficient / working set of commands.

Lemme know what you think. (This might be worth moving into its own PR.)

Reply:
Hmm... the thing is, it seems a bit "aggressive" to install anything to the global npm directory, it's not necessarily something that users would expect. Did you test the difference between --global and --user in a clean environment (with/without nvm?)

Reply:
In any case, this is something we can/should fix in a separate PR. I am restore the `--global` flag and added `npm config set prefix $HOME` which seems to fix the issue

Reply:
I did spend some time testing with the user npm bin dir when I first wrote the `amp` task runner. Nothing I tried would allow a developer to simply invoke `amp` without using workarounds like `npx amp` or `./amp.js`.

Maybe there's a way to achieve this by having `install-amp-task-runner.js` copy the `amp` runner to the user npm bin dir and then update the system path to point to it. (I couldn't figure out how to do this reliably in a platform independent way, but I'm guessing it's possible.)

Agree with trying to fix this separately 👍